### PR TITLE
Bump ViewComponent to v3.9.0

### PR DESCRIPTION
**Pull request by benilovj**
This addresses an XSS vector https://github.com/ViewComponent/view_component/pull/1950

### CODE REVIEW

**Comment by benilovj** [comment]
@peteryates 

This is in response to https://github.com/ViewComponent/view_component/security/advisories/GHSA-wf2x-8w6j-qw37

My commit includes hacks to get the specs to pass to potentially save you some time, but I'm not sure it's the right way to actually fix the underlying breakages. Happy to take your steer or if to close the PR if you want to do it yourself.

File: spec/components/govuk_component/header_component_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe(GovukComponent::HeaderComponent, type: :component) do
   before do
     allow_any_instance_of(GovukComponent::HeaderComponent::NavigationItem).to(
-      receive(:request).and_return(double(ActionDispatch::Request, get?: true, path: current_page))
+      receive(:request).and_return(double(ActionDispatch::Request, get?: true, format: double(html?: true), path: current_page))
```
Comment (peteryates) [inline spec/components/govuk_component/header_component_spec.rb:6]: Thanks @benilovj. Let me have a quick think about the approach for overriding the others, but maybe something like this makes it a bit more obvious what's going on 🤔 

```suggestion
      receive(:current_page?).and_wrap_original do |_original_method, path|
        path == current_page
      end
```

**Comment by benilovj** [comment]
I'm also seeing warnings in the RSpec run:

```
.........................................................................................................................................................................................................................................................................................................................................................................................................................................WARNING: The GovukComponent::TabComponent::Tab component rendered HTML-unsafe output. The output will be automatically escaped, but you may want to investigate.
WARNING: The GovukComponent::TabComponent::Tab component rendered HTML-unsafe output. The output will be automatically escaped, but you may want to investigate.
WARNING: The GovukComponent::TabComponent::Tab component rendered HTML-unsafe output. The output will be automatically escaped, but you may want to investigate.
...................................................................................................................................................................................................................................................................................................................................
```

I dunno if they've been introduced by this bump?

**Comment by peteryates** [comment]
They have, just tracking them down now.

**Comment by peteryates** [comment]
Those warnings can be fixed with:

```patch
diff --git a/app/components/govuk_component/tab_component.rb b/app/components/govuk_component/tab_component.rb
index 1ac03dd..8dc2250 100644
--- a/app/components/govuk_component/tab_component.rb
+++ b/app/components/govuk_component/tab_component.rb
@@ -23,7 +23,7 @@ private

     def initialize(label:, text: nil, classes: [], html_attributes: {})
       @label = label
-      @text  = text
+      @text  = h(text)

       super(classes: classes, html_attributes: html_attributes)
     end
```

Happy with the changes, once the above is in I'll get it merged and a release ready 🙂  Thanks @benilovj

**Review by peteryates — APPROVED** [review]

**Pull request merged**